ui/app/common/offline/services/offlineSyncService.js

```diff
@@ -97,6 +118,68 @@ angular.module('bahmni.common.offline')
+                    var synced = syncedInfo[dbName] || [];
+                    return $http.get(Bahmni.Common.Constants.preprocessedOfflineConceptsFilesUrl + "offline-concepts").then(function (response) {
+                        return getOfflineConceptsDataForFiles(getRemainingFileNames(response.data, synced), 0, null, dbName).then(function (uuid) {
+                            eventLogUuid = uuid;
+                            return eventLogUuid;
+                        });
+                    }).catch(function () {
+                        endSync(-1);
+                        return defer.reject();
+                    });
+                });
+            };
+            var getOfflineConceptsDataForFiles = function (fileNames, count, eventLogUuid, dbName) {
+                if (count !== fileNames.length) {
+                    return $http.get(Bahmni.Common.Constants.preprocessedOfflineConceptsUrl + fileNames[count]).then(function (response) {
+                        updatePendingEventsCount("offline-concepts", response.data.offlineconcepts.length);
+                        var lastReadEventUuid = response.data.lastReadEventUuid;
+                        return saveMetaData(response.data.offlineconcepts, 0).then(function () {
+                            updateSyncedFileNames(fileNames[count], dbName);
+                            return getOfflineConceptsDataForFiles(fileNames, ++count, lastReadEventUuid, dbName);
+                        });
+                    });
+                }
+                return $q.when(eventLogUuid);
+            };
+
+            var saveAddressHierarchyDataFromFile = function () {
+                var defer = $q.defer();
+                return getDbName().then(function (dbName) {
+                    var eventLogUuid;
+                    var syncedInfo = offlineService.getItem("synced") || {};
+                    var synced = syncedInfo[dbName] || [];
+                    return $http.get(Bahmni.Common.Constants.preprocessedAddressHierarchyFilesUrl + "AddressHierarchy").then(function (response) {
+                        return getAddressHierarchyDataForFiles(getRemainingFileNames(response.data, synced), 0, null, dbName).then(function (uuid) {
+                            eventLogUuid = uuid;
+                            return eventLogUuid;
+                        });
+                    }).catch(function () {
+                        endSync(-1);
+                        return defer.reject();
+                    });
+                });
+            };
+            var getAddressHierarchyDataForFiles = function (fileNames, count, eventLogUuid, dbName) {
+                if (count !== fileNames.length) {
+                    return $http.get(Bahmni.Common.Constants.preprocessedAddressHierarchyUrl + fileNames[count]).then(function (response) {
+                        updatePendingEventsCount("addressHierarchy", response.data.addressHierarchy.length);
```

Comment:
@gopikrishna-yaramothu can we have the constants defined instead of hardcoding the names here like "addressHierarchy", "eventLogCategories", "forms", "offline-concepts"..etc